tests/strategies/test_sharded_strategy.py

```diff
@@ -16,6 +16,9 @@
     from fairscale.nn.data_parallel.sharded_ddp import ShardedDataParallel
 
 
+# TODO(akihironitta): Ignore the warning for now since lr_schedulers doesn't wrap new OSS optimizers that are
+# initialized by the strategy.
+@pytest.mark.filterwarnings(r"ignore:Detected call of `lr_scheduler.step\(\)` before `optimizer.step\(\)`:UserWarning")
```

Comment:
After `DDPShardedStrategy` creates new OSS optimizers at:
https://github.com/PyTorchLightning/pytorch-lightning/blob/5da065e287b44e2c1fe4f7951003813ed45365c9/pytorch_lightning/strategies/sharded.py#L71

lr_schedulers never get reinitialized , and thus, `lr_scheduler.optimizer._step_count` isn't incremented.

---

I think we should reinit lr_schedulers, too, after OSS optimizers are created in order to wrap the OSS optimizers' step method with a step counter at: https://github.com/pytorch/pytorch/blob/539acb29cd7a203b614034b5efd9f5144ff89ae9/torch/optim/lr_scheduler.py#L47-L72